src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,369 @@
+      getStatesCheckCommand()
+    ).until(() -> !FaultReporter.getInstance().getFaults(SUBSYSTEM_NAME).isEmpty())
+    .andThen(Commands.runOnce(this::turnIntakeOff))
+    .withName(SUBSYSTEM_NAME + "SystemCheck");
+  }
+
+
+  private Command getVelocitiesCheckCommand() {
+    return Commands.parallel(
+      Commands.run(this::runIntakeStateMachine),
+      Commands.waitSeconds(1).andThen(
+        Commands.runOnce(() -> {
+          for (int i = 0; i < 4; i++) {
+            checkMotorVelocity(intakeMotors[i]);
+          }
+        })
+      )
+    );
+  }
+
+  private void checkMotorVelocity(IntakeMotor motor) {
+    if (motor == IntakeMotor.RIGHT_ROLLER) {
+      if (inputs.rightRollerVelocityRotationsPerSecond 
+      - inputs.rightRollerReferenceVelocityRPS 
+      > IntakeConstants.ROLLER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Right Roller Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.LEFT_ROLLER) {
+      if (inputs.leftRollerVelocityRotationsPerSecond 
+      - inputs.leftRollerReferenceVelocityRPS 
+      > IntakeConstants.ROLLER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Left Roller Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.DRUM) {
+      if (inputs.drumVelocityRotationsPerSecond 
+      - inputs.drumReferenceVelocityRPS 
+      > IntakeConstants.DRUM_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Drum Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.KICKER) {
+      if (inputs.kickerVelocityRotationsPerSecond 
+      - inputs.kickerReferenceVelocityRPS 
+      > IntakeConstants.KICKER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Kicker Too Slow"
+        );
+      }
+    }
+  }
+
+  private Command getStatesCheckCommand() {
+    IntakeState[] desiredStateSequence = {
+      IntakeState.EMPTY,
+      IntakeState.NOTE_IN_INTAKE,
+      IntakeState.IN_BETWEEN_INTAKE_AND_DRUM,
+      IntakeState.NOTE_IN_DRUM,
+      IntakeState.IN_BETWEEN_DRUM_AND_KICKER,
+      IntakeState.NOTE_IN_KICKER
+    };
+    
+    ArrayList<IntakeState> actualStateSeqeunce = new ArrayList<>();
+
+    return Commands.parallel(
+      Commands.run(this::runIntakeStateMachine),
+      Commands.waitSeconds(1).andThen(
+        Commands.runOnce(() -> checkStateSequence(actualStateSeqeunce, desiredStateSequence))
+      )
+    );
+  }
+
+  private void checkStateSequence(ArrayList<IntakeState> actualStateSequence, IntakeState[] desiredStateSequence) {
+    actualStateSequence.add(IntakeState.EMPTY);
+    IntakeState mostRecentIntakeState = IntakeState.EMPTY;
+
+    boolean correct = actualStateSequence.get(0).equals(desiredStateSequence[0]);
+
+    while (correct) {
```

Comment:
If we loop in this method, we will hang the command loop and periodic will never run and states will never change. Instead, this method will be called multiple times and, each time, it needs to check if there has been a state change. To do that mostRecentIntakeState will need to be an instance variable instead of a local variable.